### PR TITLE
DOC-4352: Developer guide for OnlyOffice connector

### PR DESCRIPTION
**Pull request by mzorai**

### CODE REVIEW

File: docs/Dev_eXo_Addons.rst [hunk 1]
```diff
@@ -1327,6 +1327,67 @@ To add an event listener using one of listed events above, you can add the follo
...
+with these parameters:
+
+-  ``path``: Defines the path to the created document's template. 
+-  ``label``: The template's label. 
+-  ``mimeType``: The document's mimetype.
+
+.. tip:: -  The ``path`` should be defined as it is made in eXo Platform components i.e. when it is preceeded
+            by **classpath** it means the template should be found on jars and when it is preceeded by 
+            **war:** it means the template should be found under webapps.
+         -  The parameter ``label`` is the traduction key for the label to be displayed in the ``New Document``
```
Comment (plamarque) [inline docs/Dev_eXo_Addons.rst]: translation

File: docs/Dev_eXo_Addons.rst [hunk 2]
```diff
@@ -1327,6 +1327,67 @@ To add an event listener using one of listed events above, you can add the follo
...
+		</external-component-plugins>  
+		
+Each new document templated to add should be defined by this object ``org.exoplatform.onlyoffice.documents.NewDocumentType``
+with these parameters:
+
+-  ``path``: Defines the path to the created document's template. 
+-  ``label``: The template's label. 
+-  ``mimeType``: The document's mimetype.
+
+.. tip:: -  The ``path`` should be defined as it is made in eXo Platform components i.e. when it is preceeded
```
Comment (plamarque) [inline docs/Dev_eXo_Addons.rst]: the meaning of prefixes is not related to this feature. so it should not documented here. please check where it is documented (or do it around kernel or extension chapters) and then replace this lengthy tip by a shorter sentence with link to the right location. Also it's better to give a concrete example ti let developer understand where how/where he should package his template files

File: docs/Dev_eXo_Addons.rst (hunk 1 above)
Comment (plamarque) [reply docs/Dev_eXo_Addons.rst]: here also, we should link to a chapter talking about translations management (hopefully it exists) and explain in which file(s) the translation property should be set (with an example)

**Review by plamarque — CHANGES_REQUESTED** [review]

File: docs/Dev_eXo_Addons.rst [hunk 3]
```diff
@@ -1327,6 +1327,67 @@ To add an event listener using one of listed events above, you can add the follo
...
+      
+      
+.. _PLFDevGuide.eXoAdd-ons.eXoOnlyOfficeConnector:
+
+==============================
+OnlyOffice connector for eXo
+==============================
+
+With :ref:`OnlyOffice connector <OnlyOffice>`, it is possible to add new document types to the
+form ``New document`` by using the :ref:`extension mechanism <PLFDevGuide.eXoAdd-ons.PortalExtension.Mechanism>`.
```
Comment (thomasdelhomenie) [inline docs/Dev_eXo_Addons.rst:1339]: Can we add a link to the "New document" chapter in the user guide ?

File: docs/Dev_eXo_Addons.rst [hunk 4]
```diff
@@ -1327,6 +1327,67 @@ To add an event listener using one of listed events above, you can add the follo
...
+					</value>
+			</collection>
+			</field>
+				</object>
+			</object-param>
+			</init-params>
+			</component-plugin>
+		</external-component-plugins>  
+		
+Each new document templated to add should be defined by this object ``org.exoplatform.onlyoffice.documents.NewDocumentType``
```
Comment (thomasdelhomenie) [inline docs/Dev_eXo_Addons.rst]:
```suggestion
Each new document template to add should be defined by this object ``org.exoplatform.onlyoffice.documents.NewDocumentType``
```

File: docs/Dev_eXo_Addons.rst [hunk 5]
```diff
@@ -1327,6 +1327,67 @@ To add an event listener using one of listed events above, you can add the follo
...
+
+-  ``path``: Defines the path to the created document's template. 
+-  ``label``: The template's label. 
+-  ``mimeType``: The document's mimetype.
+
+.. tip:: -  The ``path`` should be defined as it is made in eXo Platform components i.e. when it is preceeded
+            by **classpath** it means the template should be found on jars and when it is preceeded by 
+            **war:** it means the template should be found under webapps.
+         -  The parameter ``label`` is the traduction key for the label to be displayed in the ``New Document``
+            form. To define it, you should configure the parameter ``UINewDocumentForm.label.option``.
```
Comment (thomasdelhomenie) [inline docs/Dev_eXo_Addons.rst]: Why not keeping these explanations in the first list directly ?

Comment (thomasdelhomenie) [inline docs/Dev_eXo_Addons.rst]:
```suggestion
            form. To define it, you should configure the key ``UINewDocumentForm.label.option.{label}``.
```

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Dev_eXo_Addons.rst (hunk 4 above)
Comment (plamarque) [reply docs/Dev_eXo_Addons.rst]: now I understand :-)

File: docs/Dev_eXo_Addons.rst (hunk 3 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst:1339]: OK

File: docs/Dev_eXo_Addons.rst (hunk 5 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst]: OK

File: docs/Dev_eXo_Addons.rst (hunk 1 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst]: done

File: docs/Dev_eXo_Addons.rst (hunk 3 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst:1339]: done

File: docs/Dev_eXo_Addons.rst (hunk 5 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst]: done

File: docs/Dev_eXo_Addons.rst (hunk 2 above)
Comment (mzorai) [reply docs/Dev_eXo_Addons.rst]: Will be handled in https://jira.exoplatform.org/browse/DOC-4353

**Review by thomasdelhomenie — APPROVED** [review]

**Review by plamarque — APPROVED** [review]
LGTM

**Pull request merged**